main.go

```diff
@@ -223,3 +236,33 @@ func makeHTTPRequestHandler(watchdogConfig config.WatchdogConfig) func(http.Resp
 
 	}
 }
+
+func listenUntilShutdown(shutdownTimeout time.Duration, s *http.Server) {
+	idleConnsClosed := make(chan struct{})
+	go func() {
+		sig := make(chan os.Signal, 1)
+		signal.Notify(sig, syscall.SIGTERM)
+
+		<-sig
+
+		log.Printf("SIGTERM received.. shutting down server")
+
+		acceptingConnections = false
+
+		if err := s.Shutdown(context.Background()); err != nil {
```

Comment:
It might be a good idea to make use of a context with timeout. `ctx, _ := context.WithTimeout(context.Background(), 5*time.Second)`. As this can be used in the Shutdown, without the need of an additional construct

Reply:
Sounds good @Templum